service/controller/chartconfig/chart_config.go

```diff
@@ -44,43 +39,24 @@ func NewChartConfig(config Config) (*ChartConfig, error) {
 	if config.Fs == nil {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Fs must not be empty", config)
 	}
-	if config.G8sClient == nil {
-		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
-	}
 	if config.K8sClient == nil {
 		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
 	}
-	if config.K8sExtClient == nil {
-		return nil, microerror.Maskf(invalidConfigError, "%T.K8sExtClient must not be empty", config)
-	}
 	if config.Logger == nil {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
 
-	var crdClient *k8scrdclient.CRDClient
+	// Create a new k8sClient so the legacy scheme can be configured.
```

Comment:
This is a workaround for https://github.com/giantswarm/k8sclient/pull/21#discussion_r352067085 because the controllers use different schemes.

I think we should change to a slice as proposed by Mr @kopiczko. I'd like to work on that tomorrow if there are no objections?

cc @sslavic @tuommaki

Reply:
No objections. Makes sense.

Reply:
You can also set it like:

```
AddToScheme: func(s *runtime.Scheme) error {
	err := corev1alpha1.AddToScheme(s)
	if err != nil {
		return microerror.Mask(err)
	}
	err = applicationv1alpha1.AddToScheme(s)
	if err != nil {
		return microerror.Mask(err)
	}

	return nil
},
```

Reply:
Pawel thanks I like that a lot. Done.